cumulus/tasks/pdr-status-check/index.js

```diff
@@ -137,28 +139,36 @@ function buildOutput(event, groupedExecutions) {
  * @returns {Promise.<Object>} - an object describing the status of Step
  *   Function executions related to a PDR
  */
-function checkPdrStatuses(event) {
+async function checkPdrStatuses(event) {
   const runningExecutionArns = event.input.running || [];
 
-  const promisedExecutionDescriptions = runningExecutionArns.map((executionArn) =>
-    aws.sfn().describeExecution({ executionArn }).promise());
-
-  return Promise.all(promisedExecutionDescriptions)
-    .then(groupExecutionsByStatus)
-    .then((groupedExecutions) => {
-      const counter = getCounterFromEvent(event) + 1;
-      const exceededLimit = counter >= getLimitFromEvent(event);
+  const executions = [];
+  for (const executionArn of runningExecutionArns) {
```

Comment:
The style guide recommends avoiding iterators when possible.  Instead of a `for` loop here, consider using `map`.

http://airbnb.io/javascript/#iterators--nope

Because of the special case where the execution does not yet exist, I would probably create a `describeExecutionStatus` function and use that in the map.  Something like:

```
function describeExecutionStatus(executionArn) {
  return aws.sfn().describeExecution({ executionArn }).promise()
    .catch((e) => {
      if (e.code === 'ExecutionDoesNotExist') {
        return { executionArn: executionArn, status: 'RUNNING' };
      }
      throw e;
    });
}
```

With that function, you could then remove the loop and just say
```
const executions = Promise.all(promisedExecutionStatuses);
```

Reply:
Thank you!  I realized we should keep Promise.all after I started to look at CUMULUS-304.  I will do this under CUMULUS-304: Add AWS API throttling to pdr-status-check task.

Reply:
if we don’t use a for loop here we should consider using p-limit or we will quickly hit the step function api limit.

Reply:
@scisco  I will fix it under CUMULUS-304.